app/Http/Controllers/registrationController.php

```diff
@@ -3,11 +3,98 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use App\Models\Users;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Redis;
+use App\Models\tran;
+use Illuminate\Support\Facades\Redirect;
 
 class registrationController extends Controller
 {
-    public function index()
+    public function registerShow()
     {
         return view('form');
     }
+    public function loginShow()
+    {
+        return view('login');
+    }
+    public function register()
+    {
+        return redirect('welcome');
+        
+    }
+    public function view(){
+        $people = Users::latest('updated_at')->get();
+        $data = compact('people');
+        return view('customer-view')->with($data);
+    }
+    public function log(Request $request)
+    {
+        $request->validate([
+            'email' => 'required | email',
+            'pass' => 'required | min:5',
+        ]);
+
+        $userData = Users::where('email','=',$request->email)->first();
+
+        if(!$userData){
+            return back()->with('fail',"Email Not Found");
+        }
+        else
+        {
+            if($request->password == $userData->password)
+            {
+                $request->session()->put('logged_user',$userData->id);
+                return redirect('/welcome');
```

Comment:
```suggestion
                return redirect()->route('welcome');
```
Instead of using direct urls for routes, recommended is to define the route name and use Laravel redirect to named route. So, in case of any url updates, less code refactors needed to be performed.